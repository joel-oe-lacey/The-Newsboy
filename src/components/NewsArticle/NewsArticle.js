import React from 'react';
import './NewsArticle.scss';

const NewsArticle = ({article: {img, headline, description, url}}) => {
    return (<article className="news-article">
        <img src={img} alt={headline}></img>
        <section className="news-content">
            <h1>{headline}</h1>
            <p>{description}</p>
        </section>
        <button><a href={url}>Link to the article  -></a></button>
    </article>
    )}

export default NewsArticle;