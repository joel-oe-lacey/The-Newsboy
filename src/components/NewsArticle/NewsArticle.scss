.news-article {
    display: flex;
    flex-direction: column;
    justify-content: space-between;
    background-color: white;
    margin: 1%;
    width: 30%;
    height: 30%;

    img {
      height: 60%;
      width: 100%;
    }
}