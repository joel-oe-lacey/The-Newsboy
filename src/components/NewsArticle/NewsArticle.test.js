import React from 'react';
import { shallow } from 'enzyme';
import NewsArticle from './NewsArticle';

describe('NewsArticle', () => {
    it('renders an article utilizing available data', () => {
        const wrapper = shallow(<NewsArticle 
            key={1}
            article={{
              img:'www.imgTestUrl.com',
              headline:'Important Headline',
              description:'Important description text',
              url:'www.testRedirect.com'
            }} />)

        expect(wrapper).toMatchSnapshot();
    })
});