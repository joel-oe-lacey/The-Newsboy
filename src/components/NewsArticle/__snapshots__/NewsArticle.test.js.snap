// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewsArticle renders an article utilizing available data 1`] = `ShallowWrapper {}`;
